doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+ - 3 values: (A,B,C).
+</dd>
+
+<dt>Rogue Parameter</dt>
+<dd>
+A Parameter that is not contained by any configuration.
+<dd>
+</dl>
+
+## Configuration
+
+A PF **MUST** offer configurations as described in the Definition chapter.
+<nb>rule based parameter engine does not manipulate directly values, it
+applies configuration on the parameters.</nb>
+<why>This is what the reference implementation does.</why>
+
+Each configuration **MUST** be associated with a predicate that condition its
+eligibility. A configuration with a predicate that evaluates to `true` is called
+an "eligible configuration"
+<why>This is what the reference implementation does.</why>
+
+It **SHOULD** be possible to express a predicate to always evaluates to `true`.
+<why>in order to have parameters set to constant values or have a fallback
+configuration in a domain -- see below.</why>
+
+The predicate **SHOULD** be a "selection criterion rule". See next chapter for a
+definition.
+<why>The reference implementation uses a boolean expression based engine.</why>
+
+## Selection criterion
+
+A selection criterion **MUST** have one, and only one, state at a given time.
+
+A selection criterion **MUST** know at construction all possible states.
+<why>To be able to validate: -- rules on start -- state changes</why>
+
+The selection criterion possible states **MUST** be specifiable by directly a
+state set (`Input -> states == identity`)
+<nb>called **exclusive criterion**</nb>
+<nb>An empty set is not allowed as the criterion could not have a state.</nb>
+<why>any criterion can be crated with this one.</why>
+
+The selection criterion possible states **SHOULD** be specifiable by a combination
+of values
+<nb>combination in the mathematical sense `"ab" -> ["", "a", "b", "ab"]`</nb>
+<nb>called **inclusive criterion**</nb>
+<nb>An empty value set is allowed as its combination -- a set containing the
+empty set -- would not be empty. The empty set would be the only possible
+criteria state.</nb>
+<why>The reference implementation supports it.</why>
+
+The PF **MUST NOT** limit the number of criteria.
+
+The PF **MUST NOT** limit the number of states of any given criterion
+<ko>The reference implementation only supports 32 values for an inclusive
+criterion and 2^32 values for an exclusive criterion</ko>
+
+### Definitions
+<dl>
+<dt>Selection criterion rule</dt>
+<dd>
+Function (in the mathematical sense) that **MUST** given selection criteria
+return a Boolean.
+</dd>
+
+<dt>Rule</dt>
+<dd>
+A Boolean expression of Selection criterion rules.
+<nb>implementation only allows AND and OR combination</nb>
+<dd>
+</dl>
+
+### Criterion change
+
+The API to change criterion values **MUST** allow atomicity regarding
+configuration application. I.e. it **MUST** be possible to change multiple
+criterion values without triggering a configuration application.
+<why>Two criterion might have an excluding state. If configuration application
+was triggered after each</why>
```

Comment:
The user may describe a system's state change by parts (e.g. "X changes from 1 to 2 and Y changed from "foo" to "bar") but may not want the inference engine to recompute the configurations for each partial change.

It puts a requirement on the Parameter Framework: the user must be able to decide when the recomputing takes place --> "atomicity regarding configuration application".